Add blog post 'Anatomy of a Serverless Application'

rupakg opened the pull request.

alexdebrie commented on posts/2017-07-24-anatomy-of-a-serverless-app.md:
````diff
@@ -0,0 +1,669 @@
+$ curl -i -X POST -d '{"to_address":""}' https://yt1i5ydiu4.execute-api.us-east-1.amazonaws.com/prod/email
+
+{"message":"Bad input data or missing email address.","input":"{\"to_address\":\"\"}"}
+```
````
These four examples are pretty similar -- maybe condense to 2?

alexdebrie commented on posts/2017-07-24-anatomy-of-a-serverless-app.md:
```diff
@@ -0,0 +1,669 @@
+...
+END RequestId: 0eb38aeb-6d9d-11e7-9787-2996c7e3fdb0
+REPORT RequestId: 0eb38aeb-6d9d-11e7-9787-2996c7e3fdb0	Duration: 339.84 ms	Billed Duration: 400 ms 	Memory Size: 1024 MB	Max Memory Used: 41 MB
+
```
It's nice to show how unhandled errors manifest themselves w/ Lambda, but don't know that we want to encourage this. Maybe add a section to refactor and catch that error and return something meaningful to the user?

alexdebrie commented on posts/2017-07-24-anatomy-of-a-serverless-app.md:
```diff
@@ -0,0 +1,669 @@
+  * `frontend` folder holds the frontend application
+  * `services` folder holds the serverless service(s)
+
+A structure like this provides clear separation for the non-serverless and serverless code for the overall application. The way the non-serverless portion of the application is written is totally your choice. In this post, we will primarily focus on the serverless portion of the application inside the `services` folder.
```
The `frontend` part is confusing to me since it's not referenced in the post at all.

rupakg commented:
@alexdebrie Thanks for your feedback. I have updated the post based on your comments, fixed typos, and refactored code to add exception handling at the end.

rupakg commented:
@andrea-p Thanks for the review. Nice edits!

andrea-p commented:
schedule(07/27/2017 6:30 am)

DavidWells commented:
✅ Post scheduled for **07/27/2017 at 6:30am (America/Los_Angeles)**

**To unscheduled the post:**

1. **[Delete the previous comment ⬆](https://github.com/serverless/blog/pull/223#issuecomment-318181624)** with schedule(time) information by clicking the ❌
  ```
  ┌──────────────────────────────────────────────┐
  │                                           X  │
  ├───────────────────────────────────────────▲──┤
  │                                           │  │
  │  schedule(mm/dd/yyyy h:mm A)   ┌────────┐ │  │
  │                                │ Delete ├─┘  │
  │                                └────────┘    │
  └──────────────────────────────────────────────┘
  ```
2. (Optionally) Delete this comment to keep things tidy

> Powered by the ⚡️ [Serverless post scheduler](https://github.com/serverless/post-scheduler) | Made with ❤️  by [Serverless](https://serverless.com/)

DavidWells commented:
🎉 **The post has been published**

> Powered by the ⚡️ [Serverless post scheduler](https://github.com/serverless/post-scheduler) | Made with ❤️  by [Serverless](https://serverless.com/)
